pyartcd/pyartcd/jenkins.py

```diff
@@ -1,71 +1,107 @@
-    :param params: optional dict containing job parameters
+    If 'BUILD_URL' env var was provided, add a TRIGGERED_FROM
+    field to the build params to allow backlinking to the original job
     """
 
-    service_account = os.environ['JENKINS_SERVICE_ACCOUNT']
-    token = os.environ['JENKINS_SERVICE_ACCOUNT_TOKEN']
-
     build_url = os.environ.get('BUILD_URL', '')
     if build_url:
         params['TRIGGERED_FROM'] = build_url
+    return params
+
+
+def start_build_blocking(job_name: str, params: dict) -> str:
+    """
+    Starts a new build and blocks until it completes.
+
+    :param job_name: e.g. "build/build-sync", "scanning/scanning/images-health
+    :param params: a key-value collections to be passed to the build
+
+    Returns the build result
+    """
+
+    logger.info('Starting new build for job: %s', job_name)
+    job = jenkins.get_job(job_name)
+    build = job.invoke(build_params=params).block_until_building()
+    logger.info('Started new build for job %s',
+                build.baseurl.replace(constants.JENKINS_SERVER_URL, constants.JENKINS_UI_URL))
+
+    logger.info('Waiting for build to complete...')
+    build.block_until_complete()
+    logger.info(f'Completed with status: {build.get_status()}')
+    return build.get_status()
+
+
+def start_build_non_blocking(job_name: str, params: dict):
+    """
+    Starts a new build in "fire-and-forget" fashion
+
+    :param job_name: e.g. "build/build-sync", "scanning/scanning/images-health
+    :param params: a key-value collections to be passed to the build
+    """
+
+    logger.info('Starting new build for job: %s', job_name)
+    job = jenkins.get_job(job_name)
+    build = job.invoke(build_params=params).block_until_building()
+    logger.info('Started new build for job %s',
+                build.baseurl.replace(constants.JENKINS_SERVER_URL, constants.JENKINS_UI_URL))
+
 
-    # Build authorization header
-    auth = base64.b64encode(f'{service_account}:{token}'.encode()).decode()
-    auth_header = f'Basic {auth}'
-    headers = {'Authorization': auth_header}
-
-    # Build url
-    # If the job to be triggered is parametrized, use 'buildWithParameters' and send HTTP data
-    # Otherwise, just call the 'build' endpoint with empty data
-    url = f'https://buildvm.hosts.prod.psi.bos.redhat.com:8443/{job_path}'
-    if params:
-        url += '/buildWithParameters'
-    else:
-        url += '/build'
-
-    # Call API endpoint
-    logger.info('Triggering remote job /%s', job_path)
-    async with aiohttp.ClientSession(headers=headers) as session:
-        async with session.post(url, data=params) as response:
-            response.raise_for_status()
-            await response.text()
-
-
-async def trigger_ocp4(build_version: str):
-    await trigger_jenkins_job(
-        job_path='job/triggered-builds/job/ocp4',
+def start_ocp4(build_version: str, blocking: bool = False):
```

Comment:
Don't see a dry_run .. did we support it before? We probably are skipping triggering this if it was set in the parent.
This looks good, but I'm also thinking if there's a better way to do it instead of having 1 func for each job.

Maybe something like 
`start_build(Jobs.RHCOS, params)` 

and then we have associated param dict defined

allowed_params = {
    Jobs.RHCOS: {'BUILD_VERSION', 'NEW_BUILD', 'DRY_RUN'}
   ...
}

and then we can do some ignore case and see if params passed are in allowed_params in start_build.
What do you think?

Reply:
I could even be convinced that we don't need allowed_params, since any error / silly thing
we do in calling start_build(..) would be apparent and result in obvious failure

Reply:
I prefer to have a clear interface to call each job, without having to get into details like choosing enum from `Jobs` field, etc. Each job has a calling interface with separate params, and will be maintained independently from the others

Reply:
I'm okay with that :+1: